src/transformers/models/owlvit/processing_owlvit.py

```diff
@@ -0,0 +1,164 @@
+
+            if isinstance(text, List) and isinstance(text[0], List):
+                encodings = []
+
+                # Maximum number of queries across batch
+                max_num_queries = max([len(t) for t in text])
+
+                # Pad all batch samples to max number of text queries
+                for t in text:
+                    if len(t) != max_num_queries:
+                        t.extend([""]*(max_num_queries - len(t)))
+                        encoding = self.tokenizer(t, return_tensors=return_tensors, **kwargs)
+                        encodings.append(encoding)
+                    else:
+                        encoding = self.tokenizer(t, return_tensors=return_tensors, **kwargs)
+                        encodings.append(encoding)        
+
+            output = encodings[0]
+
+            if return_tensors == "np":
+                input_ids = [np.expand_dims(encoding["input_ids"], axis=0) for encoding in encodings]
+                input_ids = np.concatenate(input_ids)
+
+                attention_mask = [np.expand_dims(encoding["attention_mask"], axis=0) for encoding in encodings]
+                attention_mask = np.concatenate(attention_mask)
+
+            elif return_tensors == "jax":
+                input_ids = [jnp.expand_dims(encoding["input_ids"], axis=0) for encoding in encodings]
+                input_ids = jnp.concatenate(input_ids)
+
+                attention_mask = [jnp.expand_dims(encoding["attention_mask"], axis=0) for encoding in encodings]
+                attention_mask = jnp.concatenate(attention_mask)
+
+            elif return_tensors == "pt":
+                import torch
+                input_ids= [encoding["input_ids"].unsqueeze(0) for encoding in encodings]
+                input_ids = torch.cat(input_ids)
+
+                attention_mask= [encoding["attention_mask"].unsqueeze(0) for encoding in encodings]
+                attention_mask = torch.cat(attention_mask)
+            else:
+                import tensorflow as tf
+                input_ids = [tf.expand_dims(encoding["input_ids"], axis=0) for encoding in encodings]
+                input_ids = tf.concat(input_ids, axis=0)
+
+                attention_mask = [tf.expand_dims(encoding["attention_mask"], axis=0) for encoding in encodings]
+                attention_mask = tf.concat(attention_mask, axis=0)
+
+            output["input_ids"] = input_ids
+            output["attention_mask"] = attention_mask
+
+        if images is not None:
+            image_features = self.feature_extractor(images, return_tensors=return_tensors, **kwargs)
+
+        if text is not None and images is not None:
+            output["pixel_values"] = image_features.pixel_values
+            return output
+        elif text is not None:
+            return output
+        else:
+            return BatchEncoding(data=dict(**image_features), tensor_type=return_tensors)
```

Comment:
You could define `output = BatchEncoding(tensor_type=return_tensors)` before `if text is not None` on L84 to simplify this a bit

```
output = BatchEncoding(tensor_type=return_tensors)
if text is not None: 
    # Get encodings
    # Get input_ids and attention_masks
    output['input_id'] = input_ids
    output['attention_mask'] = attention_mask

if images is not None:
    image_features = self.feature_extractor(images, return_tensors=return_tensors, **kwargs)
    output["pixel_values"] = image_features.pixel_values

return output
```

Reply:
Wouldn't I need to pass in the return_tensors to self.tokenizer() in either case? I'm just initializing output to hold an empty BatchEncoding instance

Reply:
Yep, you would need to in both cases (and I made mistake in the snippet above, you'd want to create output as `output=BatchEncoding()`. The simplification was more for the if/elif/else in the return. In fact, to make sure the return types are correct you could do: 

```
output = dict()
if text is not None: 
    # Get encodings
    ...
    # Get input_ids and attention_masks
    ...
    output['input_id'] = input_ids
    output['attention_mask'] = attention_mask

if images is not None:
    # Get image features
    ...
    output["pixel_values"] = image_features.pixel_values

return BatchEncoding(data=output, tensor_type=return_tensors)
```